python_modules/libraries/dagster-dg/dagster_dg/cli/dev.py

```diff
@@ -0,0 +1,215 @@
+        *_format_forwarded_option("--log-level", log_level),
+        *_format_forwarded_option("--log-format", log_format),
+        *_format_forwarded_option("--port", port),
+        *_format_forwarded_option("--host", host),
+        *_format_forwarded_option("--live-data-poll-rate", live_data_poll_rate),
+    ]
+
+    # In a code location context, we can just run `dagster dev` directly, using `dagster` from the
+    # code location's environment.
+    if dg_context.is_code_location:
+        cmd = ["uv", "run", "dagster", "dev", *forward_options]
+        temp_workspace_file_cm = nullcontext()
+
+    # In a deployment context, dg dev will construct a temporary
+    # workspace file that points at all defined code locations and invoke:
+    #
+    #     uv tool run --with dagster-webserver dagster dev
+    #
+    # The `--with dagster-webserver` is necessary here to ensure that dagster-webserver is
+    # installed in the isolated environment that `uv` will install `dagster` in.
+    # `dagster-webserver` is not a dependency of `dagster` but is required to run the `dev`
+    # command.
+    elif dg_context.is_deployment:
+        cmd = [
+            "uv",
+            "tool",
+            "run",
+            "--with",
+            "dagster-webserver",
+            "dagster",
+            "dev",
+            *forward_options,
+        ]
+        temp_workspace_file_cm = _temp_workspace_file(dg_context)
+    else:
+        exit_with_error("This command must be run inside a code location or deployment directory.")
+
+    with pushd(dg_context.root_path), temp_workspace_file_cm as workspace_file:
+        if workspace_file:  # only non-None deployment context
+            cmd.extend(["--workspace", workspace_file])
+        uv_run_dagster_dev_process = _open_subprocess(cmd)
+        try:
+            while True:
+                time.sleep(_CHECK_SUBPROCESS_INTERVAL)
+                if uv_run_dagster_dev_process.poll() is not None:
+                    raise DgError(
+                        f"dagster-dev process shut down unexpectedly with return code {uv_run_dagster_dev_process.returncode}."
+                    )
+        except KeyboardInterrupt:
+            click.secho(
+                "Received keyboard interrupt. Shutting down dagster-dev process.", fg="yellow"
+            )
+        finally:
+            # For reasons not fully understood, directly interrupting the `uv run` process does not
+            # work as intended. The interrupt signal is not correctly propagated to the `dagster
+            # dev` process, and so that process never shuts down. Therefore, we send the signal
+            # directly to the `dagster dev` process (the only child of the `uv run` process). This
+            # will cause `dagster dev` to terminate which in turn will cause `uv run` to terminate.
+            dagster_dev_pid = _get_child_process_pid(uv_run_dagster_dev_process)
+            _interrupt_subprocess(dagster_dev_pid)
```

Comment:
Can you share more about what exactly you did to come to this conclusion? What does directly interrupting mean in this context? ctrl-c in terminal or something else?

Reply:
Ok yea `uv run` must defer signal handling to its child process to allow it to do whatever complex behavior it wants. ctrl+c in terminal sends the interrupt to the whole process group so you don't end up needing to forward it. 

Process groups are not consistent across unix and windows so I think this approach of just forwarding in the finally block is a fine approach. 

Would be ideal to test this all in windows to be sure its all working as expected.

Reply:
OK, working on figuring out how to trigger the Azure pipeline for this without changing the branch name.